client/core/core.go

```diff
@@ -2932,6 +2934,61 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 	c.updateBalances(counts)
 }
 
+func (c *Core) PromptShutdown() bool {
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+	ok := true
+	for _, dc := range c.conns {
+		if dc.hasActiveOrders() {
+			ok = false
+			break
+		}
+	}
+	if !ok {
+		fmt.Print("You have active orders. Shutting down now may result in failed swaps and account penalization. " +
+			"Do you want to quit anyway? ('y' to quit, enter to abort):")
+		scanner := bufio.NewScanner(os.Stdin)
+		scan := make(chan bool)
+		go func() {
+			scan <- scanner.Scan()
+		}()
+		select {
+		case <-c.ctx.Done():
+			return ok
+		case <-scan:
+		}
+
+		promptResp := scanner.Text()
+		err := scanner.Err()
+		if err != nil {
+			fmt.Printf("input failed: %v", err)
+			return ok
+		}
+
+		switch promptResp {
+		case "y", "yes":
+			for assetID := range c.User().Assets {
+				wallet, found := c.wallet(assetID)
+				if found && wallet.connected() {
+					if err := wallet.Lock(); err != nil {
+						fmt.Println(err)
```

Comment:
`log.Errorf("error locking wallet: %v", err)`

Reply:
Since we've determined that a wallet.Lock loop like this is needed in the shutdown sequence of the `Core` instance, I think this should be removed.  I don't think it would be hard to just do this in `(*Core).Run` after `c.wg.Wait()` here: https://github.com/decred/dcrdex/blob/a98632011f4b8fdc61c10f0f5cc9d9deffa98618/client/core/core.go#L610-L612

But @JoeGruffins is right that you wouldn't prevent the app from stopping if we left this here.  And we rarely want to fmt.Print* since the logger will write to both stdout and a log file.

Reply:
After c.wg.Wait() the client has been shutdown, how we lock the wallets?

Reply:
Pretty much like you have in comments, only without the `canLockAccts` check since it it's shutting down and it no longer matters.  Did it not work?

Reply:
Well, we get errors when lock wallet
`2020-07-21 21:20:03.583 [ERR] CORE: error locking wallet: the client has been shutdown`

Reply:
Dang, that's right the `ExchangeWallet`s share the same context as the client `Core` so they start shutting down in parallel with `Core`'s shutdown.  Either we wrap the context when creating each wallet in `(*Core).CreateWallet` and shutdown the wallets manually in `(*Core).Run` only after locking, or we just go back to locking them in `PromptShutdown`.  Please do it in `PromptShutdown` as you had it for now.  Sorry to send you in the wrong direction. I think we'll revisit this so the wallets lock themselves in shutdown automatically though.

Reply:
Okay